library/src/com/google/maps/android/geoJsonLayer/GeoJsonLayer.java

```diff
@@ -0,0 +1,293 @@
+ * data.
+ */
+public class GeoJsonLayer {
+
+    private final HashMap<GeoJsonFeature, Object> mFeatures;
+
+    private final GeoJsonRenderer mRenderer;
+
+    private final GeoJsonParser mParser;
+
+    private GeoJsonPointStyle mDefaultPointStyle;
+
+    private GeoJsonLineStringStyle mDefaultLineStringStyle;
+
+    private GeoJsonPolygonStyle mDefaultPolygonStyle;
+
+    private ArrayList<LatLng> mBoundingBox;
+
+    /**
+     * Creates a new GeoJsonLayer object
+     *
+     * @param map         GoogleMap object
+     * @param geoJsonFile JSONObject to parse GeoJSON data from
+     */
+    public GeoJsonLayer(GoogleMap map, JSONObject geoJsonFile) {
+        mFeatures = new HashMap<GeoJsonFeature, Object>();
+        mDefaultPointStyle = new GeoJsonPointStyle();
+        mDefaultLineStringStyle = new GeoJsonLineStringStyle();
+        mDefaultPolygonStyle = new GeoJsonPolygonStyle();
+        mBoundingBox = null;
+        mRenderer = new GeoJsonRenderer(mFeatures, map);
+        mParser = new GeoJsonParser(geoJsonFile);
+    }
+
+    /**
+     * Creates a new GeoJsonLayer object
+     *
+     * @param map        GoogleMap object
+     * @param resourceId Raw resource GeoJSON file
+     * @param context    Context object
+     * @throws IOException   if the file cannot be open for read
+     * @throws JSONException if the JSON file has invalid syntax and cannot be parsed successfully
+     */
+    public GeoJsonLayer(GoogleMap map, int resourceId, Context context)
+            throws IOException, JSONException {
+        this(map, createJsonFileObject(context.getResources().openRawResource(resourceId)));
+    }
+
+    /**
+     * Takes a character input stream and converts it into a JSONObject
+     *
+     * @param stream Character input stream representing the GeoJSON file
+     * @return JSONObject representing the GeoJSON file
+     * @throws java.io.IOException    if the file cannot be opened for read
+     * @throws org.json.JSONException if the JSON file has poor structure
+     */
+    private static JSONObject createJsonFileObject(InputStream stream)
+            throws IOException, JSONException {
+        String line;
+        StringBuilder result = new StringBuilder();
+        // Reads from stream
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+        try {
+            // Read each line of the GeoJSON file into a string
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+                // TODO: catch something derp
+            }
+        } finally {
+            reader.close();
+        }
+        // Converts the result string into a JSONObject
+        return new JSONObject(result.toString());
+    }
+
+    /**
+     * Gets an iterator of all GeoJsonFeature elements stored
+     *
+     * @return iterator of GeoJsonFeature elements
+     */
+    public java.util.Iterator<GeoJsonFeature> getFeatures() {
```

Comment:
returning Iterable might be more useful than returning Iterator because then developers can use it in for-each loops easily